test/unit/deferred.js

```diff
@@ -253,7 +306,48 @@ test( "jQuery.Deferred.then - deferred (progress)", function() {
 			return jQuery.Deferred(function( defer ) {
 				defer.resolve( a * b );
 			});
+		}),
+		done = assert.async();
+
+	piped.progress(function( result ) {
+		return jQuery.Deferred().resolve().then(function() {
+			return result;
+		}).then(function( result ) {
+			value3 = result;
 		});
+	});
+
+	defer.progress(function( a, b ) {
+		value1 = a;
+		value2 = b;
+	});
+
+	defer.notify( 2, 3 );
+
+	piped.then( null, null, function( result ) {
+		return jQuery.Deferred().resolve().then(function() {
+			return result;
+		}).then(function() {
+			assert.strictEqual( value1, 2, "first progress value ok" );
+			assert.strictEqual( value2, 3, "second progress value ok" );
+			assert.strictEqual( value3, 6, "result of filter ok" );
+			done();
+		});
+	});
+});
+
+test( "[DIVERGENT] jQuery.Deferred.pipe - deferred (progress)", function( assert ) {
+
+	assert.expect( 3 );
+
+	var value1, value2, value3,
+		defer = jQuery.Deferred(),
+		piped = defer.pipe( null, null, function( a, b ) {
+			return jQuery.Deferred(function( defer ) {
+				defer.resolve( a * b );
+			});
+		}),
+		done = assert.async();
 
 	piped.done(function( result ) {
```

Comment:
Interesting, but a little more obtuse than the equivalent:
```js
var uploadNearCompletion = jQuery.Deferred();
upload.progress(function( advancement ) {
    if ( advancement >= 50 ) {
        uploadNearCompletion.resolve();
    }
});
```

My chief concern, though, is the non-unique nature of progress handler execution, which can lead to implicit races. But upon more thought, I don't even think this concept is compatible with standard Promises because they exclusively bind to the first exit value from a handler associated with their creation. I **do**, however, need to add some tests and code supporting thenable return from progress handlers, e.g.:
```js
var trigger = jQuery.Deferred().notify();
trigger.then( null, null, function() {
    var notifier = jQuery.Deferred().notify( "foo" );
    setTimeout(function() {
        notifier.notify( "bar" ).resolve( "swallowed" );
    });
    return notifier;
})
    .then( fail, fail, expectFooThenBarThenFooThenBar );
trigger.notify();
```

Thanks for the insight.